hledger-lib/Hledger/Data/Types.hs

```diff
@@ -733,18 +735,31 @@ nullaccountdeclarationinfo = AccountDeclarationInfo {
 
 -- | An account, with its balances, parent/subaccount relationships, etc.
 -- Only the name is required; the other fields are added when needed.
-data Account = Account {
-   aname                     :: AccountName    -- ^ this account's full name
+data Account a = Account {
+   aname                     :: AccountName         -- ^ this account's full name
   ,adeclarationinfo          :: Maybe AccountDeclarationInfo  -- ^ optional extra info from account directives
   -- relationships in the tree
-  ,asubs                     :: [Account]      -- ^ this account's sub-accounts
-  ,aparent                   :: Maybe Account  -- ^ parent account
-  ,aboring                   :: Bool           -- ^ used in the accounts report to label elidable parents
+  ,asubs                     :: [Account a]        -- ^ this account's sub-accounts
+  ,aparent                   :: Maybe (Account a)  -- ^ parent account
+  ,aboring                   :: Bool                -- ^ used in the accounts report to label elidable parents
   -- balance information
-  ,anumpostings              :: Int            -- ^ the number of postings to this account
-  ,aebalance                 :: MixedAmount    -- ^ this account's balance, excluding subaccounts
-  ,aibalance                 :: MixedAmount    -- ^ this account's balance, including subaccounts
-  } deriving (Generic)
+  ,abalances                 :: AccountBalances a   -- ^ historical and date-associated account balances
```

Comment:
It's hard to be clear on this everywhere, but maybe here mention both "end balances" (on a date) and "balance changes" (in a period) are represented, consistent with terminology elsewhere.

Reply:
Done